src/finite_diff.jl

```diff
@@ -0,0 +1,63 @@
+"""
+    finite_diff(J, f, fx, coloring)
+
+Compute the jacobian of function  f  at  given
+point x via central forward differencing along
+direction vectors determined by graph coloring
+"""
+function finite_diff(
+    f,
+    x::AbstractArray{<:Number},
+    coloring::AbstractArray{<:Integer})
+
+    num_sets = maximum(coloring)
+
+    fx = f(x)
+    n = length(x)
+    m = length(fx)
+
+    J = zeros(Float64, m , n)
+
+    A = rand(n,n);
+    id = diagm(0=>fill(1., size(A,1)))
+
+    ep = sqrt(eps(real(Float64)))
+
+    for i = 1:n
+        dx = x + ep.*id[:,i]
+        J[:,i] = (f(dx)-fx) / ep
+    end
+
+    return J
+
+end
+
+
+"""
+        find_vectors(coloring)
+
+Calculate the direction vectors needed for forward
+difference function from the given coloring  of  a
+graph
+"""
+function find_vectors(coloring::AbstractArray{<:Integer})
```

Comment:
You don't want to allocate like this

Reply:
Can you explain how to pass the coloring vector instead?

Reply:
Just add a keyword argument for a vector of integers for the colors

Reply:
Let me know if you want starter code

Reply:
Yeah I think that'd be of great help at the moment.